stock_picking_portal/controllers/portal.py

```diff
@@ -0,0 +1,273 @@
+        return {
+            "all": {
+                "label": _("All"),
+                "domain": [("picking_type_id.code", "in", ("outgoing", "incoming"))],
+            },
+            "outgoing": {
+                "label": _("Delivery"),
+                "domain": [("picking_type_id.code", "=", "outgoing")],
+            },
+            "incoming": {
+                "label": _("Receipt"),
+                "domain": [("picking_type_id.code", "=", "incoming")],
+            },
+        }
+
+    def _prepare_stock_operations_portal_rendering_values(
+        self,
+        page=1,
+        date_begin=None,
+        date_end=None,
+        sortby=None,
+        filterby=None,
+        **kwargs
+    ):
+        partner = request.env.user.partner_id
+        StockPicking = request.env["stock.picking"]
+        url = "/my/stock_operations"
+        domain = self._get_prepared_operation_domain(partner)
+        if not sortby:
+            sortby = "date"
+        searchbar_filters = self._get_stock_operations_searchbar_filters()
+        # default filter by value
+        if not filterby:
+            filterby = "all"
+        domain += searchbar_filters[filterby]["domain"]
+        values = self._prepare_portal_layout_values()
+        searchbar_sortings = self._get_stock_operations_searchbar_sortings()
+        sort_order = searchbar_sortings[sortby]["order"]
+
+        if date_begin and date_end:
+            domain += [
+                ("scheduled_date", ">", date_begin),
+                ("scheduled_date", "<=", date_end),
+            ]
+        pager_values = portal_pager(
+            url=url,
+            total=StockPicking.search_count(domain),
+            page=page,
+            step=self._items_per_page,
+            url_args={"date_begin": date_begin, "date_end": date_end, "sortby": sortby},
+        )
+        operations = StockPicking.search(
+            domain,
+            order=sort_order,
+            limit=self._items_per_page,
+            offset=pager_values["offset"],
+        )
+        for item in operations:
+            item._portal_ensure_token()
+        values.update(
+            {
+                "date": date_begin,
+                "stock_operation_ids": operations if operations else StockPicking,
+                "page_name": "stock_operations",
+                "pager": pager_values,
+                "default_url": url,
+                "searchbar_sortings": searchbar_sortings,
+                "sortby": sortby,
+                "searchbar_filters": OrderedDict(sorted(searchbar_filters.items())),
+            }
+        )
+        return values
+
+    @http.route(
+        ["/my/stock_operations/<int:operation_id>"],
+        type="http",
+        auth="public",
+        website=True,
+    )
+    def portal_stock_operation_page(
+        self,
+        operation_id,
+        report_type=None,
+        access_token=None,
+        message=False,
+        download=False,
+        **kw
+    ):
```

Comment:
Please add doc string for this endpoint.

Reply:
I'll fix it